cishouseholds/pipeline/pipeline_stages.py

```diff
@@ -372,6 +381,32 @@ def _inner_function(
     return _inner_function
 
 
+@register_pipeline_stage("process_soc_data")
+def process_soc_data(
+    soc_file_pattern: str,
+    survey_responses_table: str,
+    soc_coded_survey_responses_table: str,
+    unioned_soc_lookup_table: str,
+):
+    """
+    Process soc data and combine result with survey responses data
+    """
+    dfs = []
+    file_list = get_files_to_be_processed(soc_file_pattern, use_file_date=False)
```

Comment:
It might be better to call `list_contents` directly here, rather than adjust the functionality of `get_files_by_date` for example?

`get_files_to_be_processed` includes logging that prevents you from processing the same files twice (i.e. running this stage more than once). If we intend to reprocess the whole set of SOC files each time, we shouldn't be logging the processed files for this one. Using `list_contents` directly would avoid this issue